### PR TITLE
AB: "PM2.5" should be the default parameter

### PR DESCRIPTION
**Pull request by LinX64**
So about the problem, I already set the stream to set pm2.5 as the default value, but for some reason, it didn't work for mobile active sessions, now it should be setting for all sessions including mobile active, but I remember that we were facing a problem by setting only to PM2.5, that's why this needs to be fully tested.

### CODE REVIEW

**Comment by LinX64** [comment]
[mobile active > AB: "PM2.5" should be the default parameter selection, currently it's "F"](https://trello.com/c/BNVf7ROD/1675-mobile-active-ab-pm25-should-be-the-default-parameter-selection-currently-its-f)

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt [hunk 1]
```diff
@@ -104,13 +104,9 @@ class SessionPresenter() {
...
     companion object {
-        fun defaultStream(session: Session?): MeasurementStream? {
+        fun defaultPm2point5Stream(session: Session?): MeasurementStream? {
             val sortedByDetailedType = session?.streamsSortedByDetailedType()
-            val pm2point5 =
-                sortedByDetailedType?.find { it.detailedType == SensorName.PM2_5.detailedType }
-            val firstStream = sortedByDetailedType?.firstOrNull()
-
-            return pm2point5 ?: firstStream
+            return sortedByDetailedType?.find { it.detailedType == SensorName.PM2_5.detailedType }
```
Comment (czorek) [inline app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt:109]: So what will happen now if the pm2.5 stream is not found? Do we have any fallback now? Do we need it?

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt [hunk 2]
```diff
@@ -28,7 +28,7 @@ class SessionPresenter() {
         loading: Boolean = false
     ) : this() {
         this.session = session
-        this.selectedStream = selectedStream ?: defaultStream(session)
+        this.selectedStream = selectedStream ?: defaultPm2point5Stream(session)
```
Comment (czorek) [inline app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt:31]: I think I might change the name to `defaultPm2_5Stream` to have it shorter and consistent with `SensorName.PM2_5`

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt (hunk 1 above)
Comment (LinX64) [reply app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt:109]: Good point!

I thought the same, I need to do more investigation on this.

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt (hunk 2 above)
Comment (LinX64) [reply app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt:31]: Yes but lint says you cannot have an underscore in the function name, that was the reason why I added it like that.

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt [hunk 3]
```diff
@@ -104,13 +104,19 @@ class SessionPresenter() {
...
+        fun defaultPm2point5Stream(session: Session?): MeasurementStream? {
             val sortedByDetailedType = session?.streamsSortedByDetailedType()
             val pm2point5 =
                 sortedByDetailedType?.find { it.detailedType == SensorName.PM2_5.detailedType }
-            val firstStream = sortedByDetailedType?.firstOrNull()
 
-            return pm2point5 ?: firstStream
+            return if (session != null && pm2point5 != null) pm2point5 else defaultFirstStream(
+                session
+            )
```
Comment (xameYeoh) [inline app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt:119]: Does not work, and to me it looks like it does the same as previous code but it does it with less clarity

Comment (LinX64) [reply app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionPresenter.kt]: Please check the new approach with the new commit.

File: app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionViewMvcImpl.kt [hunk 4]
```diff
@@ -172,9 +176,12 @@ abstract class SessionViewMvcImpl<ListenerType>(
     }
 
     private fun bindSelectedStream() {
-        if (mSessionPresenter != null && mSessionPresenter?.selectedStream == null) {
+        val session = mSessionPresenter?.session
+
+        if (session?.hasMeasurements() == true && session.mStreamsCount() > 5) {
             mSessionPresenter?.setDefaultStream()
         }
+
```
[the remark this answers is not in the document]
Comment (xameYeoh) [reply app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionViewMvcImpl.kt:184]: There is no way it does not get called for external sessions - they are the same as local ones. This method gets called for all the session cards on all the tabs

[the remark this answers is not in the document]
Comment (xameYeoh) [reply app/src/main/java/pl/llp/aircasting/ui/view/screens/dashboard/SessionViewMvcImpl.kt:184]: As for the Second thing when you call setDefaultStream yourself - I guess you refer to the external session being bound on the bottom sheet.   So you are saying that the default stream will be selected from there already?

**Pull request merged**